src/Header/HeaderWrap.php

```diff
@@ -112,9 +112,27 @@ public static function mimeDecodeValue($value)
 
         $decodedValue = iconv_mime_decode($value, ICONV_MIME_DECODE_CONTINUE_ON_ERROR, 'UTF-8');
 
+        // imap (unlike iconv) can handle multibyte headers which are splitted across multiple line
+        if (self::isNotDecoded($value, $decodedValue) && extension_loaded('imap')) {
+            return array_reduce(
+                imap_mime_header_decode(imap_utf8($value)),
+                function ($acc, $item) {
```

Comment:
perhaps more useful variable names than `$acc` and `$item`?

Reply:
updated